invoke-ai/InvokeAI
# feat(ui): Middle mouse button (MMB) canvas panning implementation

## Summary

This PR introduces middle-mouse-button (MMB) panning functionality for the CanvasV2 stage. It leverages existing tool buffering mechanisms as well as `startDrag` and `stopDrag` events. This feature was discussed in detail on [Discord](https://discordapp.com/channels/1020123559063990373/1288038347188731969/1288058071347888158).

## Related Issues / Discussions

Closes #3042 

## QA Instructions

1. Open **CanvasV2** in **Editor** mode.
2. Use the MMB to pan around the stage.
3. Enable **Dynamic Grid** and **BBOX overlay**. Pan around to verify that no graphical artifacts are present.
4. Switch to the **Move** tool, then move any layer using the left mouse button (LMB). While the cursor is within the selected layer, pan the stage with the MMB. Ensure that the layer does not move unexpectedly.
5. Disable **snapping to grid** and pan with the MMB. Ensure that the BBOX stays aligned with the layer it surrounds.
6. Switch to **BBOX selection** mode and pan with the MMB. Verify that the BBOX remains aligned with all layers.
7. Switch to the **eyedropper** tool and select a contrasting color from your canvas. Pan with the MMB and ensure that the eyedropper's brush color does not change when the MMB is pressed.
8. While in **BBOX selection** mode, try moving the BBOX itself with the LMB.
9. During all steps, observe the active tool in the toolbar. It should switch to **View** tool while MMB is held down and revert to the previously selected tool once released.
10. Finally, attempt panning using the LMB while holding the **spacebar**. This behavior should remain unaffected.

## Merge Plan

 - N/A

## Checklist
